community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -312,7 +316,7 @@ public void markForTermination()
             if ( stillSameTransaction && canBeTerminated() )
             {
                 failure = true;
-                terminated = true;
+                terminationReason = reason;
```

Comment:
Yes, let’s tackle this as a separate issue.

Cheers,
Chris

> On 01 Jul 2016, at 09:08, Anton Persson <notifications@github.com> wrote:
> 
> In community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java <https://github.com/neo4j/neo4j/pull/7477#discussion_r69258346>:
> 
> > @@ -312,7 +316,7 @@ public void markForTermination()
> >              if ( stillSameTransaction && canBeTerminated() )
> >              {
> >                  failure = true;
> > -                terminated = true;
> > +                terminationReason = reason;
> There is another problem where a tx that is being markForTermination while being close and the termination is not scheduled until the tx has returned to the pool and then initialised again, effectively terminating a completely different tx.
> 
> Here is a branch with test that reproduce this problem in a single thread by keeping the reference to the old TopLevelTransaction which holds the pooled KernelTransactionImplementation.
> https://github.com/burqen/neo4j/tree/2.3-tx-terminate-after-reuse <https://github.com/burqen/neo4j/tree/2.3-tx-terminate-after-reuse>
> I suggest grabbing the lock earlier in markForTermination and otherwise leave this problem to be adressed separate.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/neo4j/neo4j/pull/7477/files/e50e5ab9232bbc1050ccbfb3a4fa678df8f5aa8a#r69258346>, or mute the thread <https://github.com/notifications/unsubscribe/AAAfOcsNYBJInuCw5EvpQumFXQ7RxQhzks5qRLzRgaJpZM4JBmm9>.
>